public/js/monitor.js

```diff
@@ -75,9 +55,24 @@ function showFalseDoor(){
 
 }
 
+async function sha1(message) {
+  const msgBuffer = new TextEncoder("utf-8").encode(message);
+  const hashBuffer = await crypto.subtle.digest("SHA-1", msgBuffer);
+  const hashArray = Array.from(new Uint8Array(hashBuffer));
+  const hashHex = hashArray.map(b => ("00" + b.toString(16)).slice(-2)).join("");
+  return hashHex;
+}
 
+async function hashEmailAndSend(emailFormSubmitEvent) {
+  emailFormSubmitEvent.preventDefault();
+  const emailForm = emailFormSubmitEvent.target;
+  for (const emailInput of emailForm.querySelectorAll("input[type=email]")) {
+    emailInput.value = await sha1(emailInput.value);
+  }
+  emailForm.submit();
+}
```

Comment:
This will change the value of the textbox to the hash, no? Instead, how about this:

- Change the name/id of the current email input box (or just get rid of the name/id)
- Add a hidden input with that name/id (name="email" I think)
- In this function, set the value of the hidden input to the hash, and then submit the form. This way the user won't see a hash in their UI.

Reply:
Good call.